Makes luxury shuttle more user friendly

cacogen opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

People can't seem to work out how to use the shuttle and hate it for that reason. This makes it easier for them.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Hopefully more people will be able to board the first class area of the luxury shuttle and appreciate it.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl: cacogen
add: Luxury shuttle now tells you if you don't have arms to drag money into the ticket collector, which will hopefully teach players who don't know to do this
add: Luxury shuttle can now access accounts on dragged ID cards
tweak: Change doesn't disappear if the payee isn't human or doesn't have a bank account
tweak: Change is now placed in the payee's hand, and is otherwise placed on the ground with the payee dragging it if their hands are full or they don't have any
tweak: Luxury shuttle now tells you privately how much more money you need, for less spam. It will still loudly welcome you aboard in front of your co-workers.
/:cl:

**I've edited this since these drive-by thumbs so change is placed in-hand for humans instead.**
<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

cacogen committed:
makes luxury shuttle more user friendly

FantasticFwoosh commented:
If there isn't a proletariat uprising against the captain as soon as the shuttle is bought and locked in, or people start taking additional work to pay off their fare, the shuttles's gimmick would be wasted with this PR.

Please remove the spare change feature for something else, like the heads altruistically paying off fares to assist crewmembers boarding out of their own pocket at shuttle console. :+1: for all the other changes.

Dawson1917 commented:
tweak: Luxury shuttle now drops change at the feet of people boarding for people without enough money to take

Everything is good besides this. This shuttle is, essentially, for griff and is funny for it. Locking people in the back while all the heads of staff and miners and assistants who stole from ID cards ride in the front compartment is the point. Heads paying out of their pocket for the sake of letting everyone else not ride in the mule den goes against the theme of the shuttle, in that all the heads are selfish and couldn't care less for the rest of the crew's conditions so long as they get their champagne and caviar.

bgobandit commented:
space communism :+1:

cacogen commented:
>If there isn't a proletariat uprising against the captain as soon as the shuttle is bought and locked in, or people start taking additional work to pay off their fare, the shuttles's gimmick would be wasted with this PR.

This doesn't happen

>Please remove the spare change feature for something else

I'll put it in their hands instead, if only for the people giving this downward facing thumbs

cacogen committed:
Makes it so change is placed in-hand, slimes aren't told to pull something when they can't and dragged IDs override worn IDs so a bank account is looked for on them instead

cacogen committed:
Makes it so if the change doesn't go into a human's hands, they will be pulling it behind them instead.

Farquaar commented:
I'd echo the sentiments of others here. It's more fun when players actively have to give their fellow crewmembers money if they want to the poor working stiffs to ride in the luxury area. It facilitates more player interaction and roleplay that way.

actioninja commented on code/modules/shuttle/special.dm:
```diff
@@ -262,40 +284,50 @@
-			var/datum/bank_account/account = H.get_bank_account()
+	if(payees[AM] < threshold)
+		var/armless
+		if(!ishuman(AM) && !istype(AM, /mob/living/simple_animal/slime))
```
Is this intended to be not slime?

cacogen replied:
Yeah. Slimes can't drag. Their drag proc is overridden. I wish there was an easier way to identify mobs that can't drag but aside from making a global list or changing how dragging is handled for all mobs I can't think of one.

actioninja replied:
Got it.
